macros/GroupTableAnswer.pl

```diff
@@ -0,0 +1,292 @@
+            //  https://stackoverflow.com/questions/1499889/remove-html-tags-in-javascript-with-regex
+            var answers_string = answers_array.join(",").replace(/(<([^>]+)>)/ig, '');
+            document.getElementById('AnSwEr_$id').setAttribute('value', answers_string);
+
+            //  mainly for debugging
+            //sessionStorage.setItem("group",document.getElementById("div_$id").innerHTML);
+        
+        }, false);
+        
+        </script>
+        );
+
+        # insert the html hidden input and check the answer
+        my $ans_id = 'AnSwEr_' . $id;
+        TEXT( NAMED_HIDDEN_ANS_RULE($ans_id) );
+        NAMED_ANS( $ans_id, $ans->cmp(ordered=>1, 'num_rows'=>$num_rows, 'num_cols'=>$num_cols)->withPostFilter(\&group_table_post_filter) ); 
+
+        return $table_html . $js;
+
+    }    # end HTML mode
+
+}    # end GroupTable
+
+
+
+sub group_table_post_filter {
+    # for answer checking, we use lists; however, we want the answer preview 
+    # and correct answer to be displayed as a table, which is what this post filter does.
+
+    my $ans_hash = shift;
+
+    my $nr = $ans_hash->{num_rows};
+    my $nc = $ans_hash->{num_cols};
+
+    #########################################
+    my $stu = $ans_hash->{student_ans};
+    my @stu_array = split(',' , $stu);
+    #warn scalar(@stu_array);
+
+    my $str = '';
+    foreach my $i (0..($nr-1)) {
+        $str .= join(','  , @stu_array[($i*$nc)..(($i+1)*$nc - 1)]) . '<br>';
+    }
+    $ans_hash->{student_ans} = $str;
+
+    #########################################
+    my $stu = $ans_hash->{preview_latex_string};
+    my @stu_array = split(',' , $stu);
+
+    my $str = '\begin{array}{';
+    $str .= "c" x $nc . '} ';
+    foreach my $i (0..($nr-1)) {
+        $str .= join(' & '  , @stu_array[($i*$nc)..(($i+1)*$nc - 1)]) . ' \cr ';
+    }
+    $str .= ' \end{array}';
+    $ans_hash->{preview_latex_string} = $str;
+
+    ######################################## 
+    my $stu = $ans_hash->{preview_text_string};
+    my @stu_array = split(',' , $stu);
+
+    my $str = '';
+    foreach my $i (0..($nr-1)) {
+        $str .= join(','  , @stu_array[($i*$nc)..(($i+1)*$nc - 1)]) . '<br>';
+    }
+    $ans_hash->{preview_text_string} = $str;
+
+    ########################################
+    my $cor = $ans_hash->{correct_ans_latex_string};
+    my @cor_array = split(',' , $cor);
+    
+    my $str = '\begin{array}{';
+    $str .= "c" x $nc . '} ';
+    foreach my $i (0..($nr-1)) {
+        $str .= join(' & '  , @cor_array[($i*$nc)..(($i+1)*$nc - 1)]) . ' \cr ';
+    }
+    $str .= ' \end{array}';
+    $ans_hash->{correct_ans_latex_string} = $str;
+
+    ######################################## 
+    my $cor = $ans_hash->{correct_ans};
+    my @cor_array = split(',' , $cor);
+
+    my $str = '';
+    foreach my $i (0..($nr-1)) {
+        $str .= join(','  , @cor_array[($i*$nc)..(($i+1)*$nc - 1)]) . '<br>';
+    }
+    $ans_hash->{correct_ans} = $str;
+
```

Comment:
Lines 263 - 282 should be taken out of this method and moved into the GroupTable method above.  You will need to make a few minor modifications to make it work there.  This shouldn't be done when the student answers are processed, but should be done when the cmp object is created.